build.gradle

```diff
@@ -62,8 +66,39 @@ dependencies {
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.2'
     testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.2'
     testImplementation "org.opensearch.test:framework:3.0.0-SNAPSHOT"
+    requireJavadoc "org.plumelib:require-javadoc:1.0.3"
+}
+
+// this task tests for the presence of javadocs but not the content/tags
+task requireJavadoc(type: JavaExec) {
+  group = 'Documentation'
+  description = 'Ensures that Javadoc documentation exists.'
+  mainClass = "org.plumelib.javadoc.RequireJavadoc"
+  classpath = configurations.requireJavadoc
+  args "src/main/java" 
+  args "--dont-require-private=true"
+  // temporary, the netty4 package will eventually be in OpenSearch main repo
+  // See https://github.com/opensearch-project/OpenSearch/issues/3118
+  args "--exclude=netty4"
+  // temporary until require-javadoc:1.0.4 allows suppressing getters and setters
```

Comment:
Can we have an issue to address this and link it here?

Reply:
Added a link to the issue other repo. It doesn't actually need to be updated unless/until we start wanting to add more exclusions and it can be handled then.